src/Theme.cpp

```diff
@@ -0,0 +1,411 @@
+    Color("TableRowBgAlt", ImGuiCol_TableRowBgAlt),
+    Color("TextSelectedBg", ImGuiCol_TextSelectedBg),
+    Color("DragDropTarget", ImGuiCol_DragDropTarget),
+    Color("NavHighlight", ImGuiCol_NavHighlight),
+    Color("NavWindowingHighlight", ImGuiCol_NavWindowingHighlight),
+    Color("NavWindowingDimBg", ImGuiCol_NavWindowingDimBg),
+    Color("ModalWindowDimBg", ImGuiCol_ModalWindowDimBg),
+};
+
+// Internal Load/Dump functions
+
+static const std::string SchemaVersionLabel = "ImGuiThemeVersion";
+static const std::string SchemaVersionValue = "1.0.0";
+
+static const std::string ThemeNameLabel = "ThemeName";
+static const std::string StyleSectionLabel = "style";
+static const std::string ColorsSectionLabel = "colors";
+
+static bool ThemeJsonFromStyle(json& aOutJson, const ImGuiStyle& aStyle)
+{
+    aOutJson = json{{SchemaVersionLabel, SchemaVersionValue}, {ThemeNameLabel, "<final merged theme in effect>"}, {StyleSectionLabel, {}}, {ColorsSectionLabel, {}}};
+
+    json& styleJson = aOutJson[StyleSectionLabel];
+
+    for (const auto& valueEncoder : StyleCodecs)
+    {
+        valueEncoder.encodeFrom(aStyle, styleJson);
+    }
+
+    json& colorJson = aOutJson[ColorsSectionLabel];
+
+    for (const auto& valueEncoder : ColorCodecs)
+    {
+        valueEncoder.encodeFrom(aStyle, colorJson);
+    }
+
+    return true;
+}
+
+static bool StyleFromThemeJson(ImGuiStyle& aOutStyle, std::ifstream& aMaybeTheme)
+{
+    const json themeJson = json::parse(aMaybeTheme, nullptr, false, true);
+
+    if (themeJson.is_discarded())
+    {
+        Log::Error("Failed to parse theme json");
+        return false;
+    }
+
+    std::string schemaVersion = themeJson.value(SchemaVersionLabel, "");
+
+    if (schemaVersion != SchemaVersionValue)
+    {
+        Log::Error("Theme schema version {} is not supported, may be invalid", schemaVersion);
+        return false;
+    }
+
+    Log::Info("Successfully parsed theme: {}", themeJson.value(ThemeNameLabel, "<unnamed theme>"));
+    Log::Info("Theme: {}", themeJson.dump(4));
+
+    json styleParameters = themeJson.value(StyleSectionLabel, json::object());
+
+    for (const auto& valueDecoder : StyleCodecs)
+    {
+        valueDecoder.decodeInto(aOutStyle, styleParameters);
+    }
+
+    json colorParameters = themeJson.value(ColorsSectionLabel, json::object());
+
+    for (const auto& valueDecoder : ColorCodecs)
+    {
+        valueDecoder.decodeInto(aOutStyle, colorParameters);
+    }
+
+    return true;
+}
+
+//
+// API
+//
+
+bool LoadStyleFromThemeJson(const std::filesystem::path& aThemePath, ImGuiStyle& aOutStyle)
+{
+    bool loadedSuccessfully = false;
+
+    const auto cThemeJsonPath = GetAbsolutePath(aThemePath, "", false);
+
+    Log::Info("Trying to load theme from {}...", UTF16ToUTF8(cThemeJsonPath.native()));
+
+    if (!cThemeJsonPath.empty())
+    {
+        std::ifstream themeJson(cThemeJsonPath);
+        if (!themeJson)
+        {
+            Log::Info("No theme file found!");
```

Comment:
for example here, you may just return false after logging the message. 
reduces nesting also nicely.